src/protocol/codecs/msgpack.rs

```diff
@@ -0,0 +1,278 @@
+use std::{
+    fmt::{self, Debug, Display, Formatter},
+    str,
+};
+
+use data_encoding::{BASE32_NOPAD, BASE64};
+use serde::{
+    de::{Error, Visitor},
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+
+type Period = u64;
```

Comment:
Docs for these types would be cool.

Reply:
Most of the comments in this file are copied from their repo.
There where you don't find comments are the locations from their repo which are missing comments 😃

I'll see if I can think of some comments for these since I'm not sure yet how to document all their stuff 💪

Reply:
Cool, just a thought; if you come across it, feel free to add later.

Reply:
No, no, it's good that you mentioned it. I invested a little time into Algorand's proposal reading. Which is also good! 👍